### PR TITLE
add create or update app apiserver

### PR DESCRIPTION
**Pull request by wonderflow**
add create or update app apiserver and tests

### CODE REVIEW

File: pkg/apiserver/rest/apis/core.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (hongchaodeng) [inline pkg/apiserver/rest/apis/core.go:1]: change the file name to app.go

File: pkg/apiserver/rest/services/catalog_test.go [hunk 2]
```diff
@@ -59,7 +59,7 @@ var _ = Describe("Test Catalog Service", func() {
 		req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
 		rec = httptest.NewRecorder()
 		c = e.NewContext(req, rec)
-		c.SetPath("/catalogs/:catalogName")
+		c.SetPath("/dashboard/catalogs/:catalogName")
```
Comment (hongchaodeng) [inline pkg/apiserver/rest/services/catalog_test.go:62]: revert this change?

File: pkg/apiserver/rest/services/application.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+// NewApplicationService create an application service
+func NewApplicationService(kc client.Client) *ApplicationService {
+	return &ApplicationService{
+		k8sClient: kc,
+	}
+}
+
+// CreateOrUpdateApplication will create or update application
+// POST /v1/namespaces/<namespace>/applications/<appname>
+func (s *ApplicationService) CreateOrUpdateApplication(c echo.Context) error {
```
Comment (hongchaodeng) [inline pkg/apiserver/rest/services/application.go:46]: Just call it PutApplication

File: pkg/apiserver/rest/rest_server.go [hunk 4]
```diff
@@ -91,13 +91,26 @@ func (s *restServer) Run(ctx context.Context) error {
 
 func (s *restServer) registerServices() error {
 
+	/* **************************************************************  */
+	/* ************   Dashboard API Route Group     *****************  */
+	/* **************************************************************  */
+
+	dashboard := s.server.Group("/dashboard")
```
Comment (wonderflow) [inline pkg/apiserver/rest/rest_server.go:98]: 都用 v1

File: pkg/apiserver/rest/rest_server.go [hunk 5]
```diff
@@ -91,13 +91,26 @@ func (s *restServer) Run(ctx context.Context) error {
...
+	dashboard.GET("/catalogs/:catalogName", catalogService.GetCatalog)
+	dashboard.DELETE("/catalogs/:catalogName", catalogService.DelCatalog)
+
+	/* **************************************************************  */
+	/* *************       Open API Route Group     *****************  */
+	/* **************************************************************  */
+
+	openapi := s.server.Group("/v1")
+	applicationService := services.NewApplicationService(s.k8sClient)
+	openapi.POST("/namespaces/:namespace/applications/:appname", applicationService.CreateOrUpdateApplication)
```
Comment (wonderflow) [inline pkg/apiserver/rest/rest_server.go:113]: `/envs/<env>`

File: pkg/apiserver/rest/apis/core.go (hunk 1 above)
Comment (wonderflow) [reply pkg/apiserver/rest/apis/core.go:1]: put all core API in one file

**Review by hongchaodeng — APPROVED** [review]
lgtm
